packages/grid/src/Grid.tsx

```diff
@@ -551,6 +552,58 @@ class Grid extends PureComponent<GridProps, GridState> {
     }
   }
 
+  setFocusRow(focusedRow: number): void {
+    this.setState({ top: focusedRow - 1 }, () => this.moveFocusRow());
+    this.setState({
+      selectedRanges: [
+        new GridRange(null, focusedRow - 1, null, focusedRow - 1),
+      ],
+    });
+  }
+
+  moveFocusRow(): void {
+    if (!this.metrics || !this.prevMetrics) {
+      return;
+    }
+
+    const { top } = this.state;
+
+    const {
+      rowHeight,
+      bottom,
+      visibleRowYs,
+      visibleRowHeights,
+      columnHeaderHeight,
+      userRowHeights,
+    } = this.metrics;
+
+    const arr: [number, number][] = [];
+    userRowHeights.forEach((x, number) => arr.push([x, number]));
+    let currentTop = top;
+    const tableHeight =
+      (this.canvas?.clientHeight ?? columnHeaderHeight) - columnHeaderHeight;
+
+    const halfViewportHeight =
+      Math.round(tableHeight / 2) - getOrThrow(visibleRowHeights, top);
+    const bottomToFill = Math.max(
+      0,
+      tableHeight -
+        rowHeight * 2 -
```

Comment:
So - `Grid` has adjustable row heights, whereas `IrisGrid` does not. Since this is in `Grid`, need to take the row heights into consideration.

Right now, you're setting the top to where you want the focus to be, and then you're moving on the next tick after the metric calculator has updated. Instead, what you should do is just use `metricCalculator.getLastTop` to calculate it, something like:
```
const top = metricCalculator.getLastTop(state, focusRow, halfTableHeight)
```

I think that should get you where you want.